app/v1/models/domain/TypeOfLoss.scala

```diff
@@ -0,0 +1,58 @@
+/*
+ * Copyright 2019 HM Revenue & Customs
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package v1.models.domain
+
+import play.api.libs.json.{ JsString, JsonValidationError, Reads, Writes }
+import v1.models.des.{ IncomeSourceType, LossType }
+
+sealed trait TypeOfLoss {
+  def isProperty: Boolean                          = false
+  def toLossType: Option[LossType]                 = None
```

Comment:
Couldn't this not be an optional? It doesn't need to have a defined value in the trait.

Reply:
Because only `uk-property-*fhl` have LossType and only `self-employment*` have IncomeSourceType.

Did toy with having separate PropertyType and SelfEmploymentType traits for these so that they would not need to be optional, but thought this might over-complicate things.